### PR TITLE
Add account restriction functions

### PR DESCRIPTION
**Pull request by TriPhan96**
- Account address restriction parser function
- Account mosaic restriction parser function
- Account operation restriction parser function

### CODE REVIEW

File: src/xym/format/format.c [hunk 1]
```diff
@@ -108,28 +108,75 @@ static void uint8_formatter(const field_t *field, char *dst) {
...
-        CASE_FIELDVALUE(XYM_TXN_ACCOUNT_KEY_LINK, "Account Key Link")
-        CASE_FIELDVALUE(XYM_TXN_NODE_KEY_LINK, "Node Key Link")
-        CASE_FIELDVALUE(XYM_TXN_VOTING_KEY_LINK, "Voting Key Link")
-        CASE_FIELDVALUE(XYM_TXN_VRF_KEY_LINK, "Vrf Key Link")
-        CASE_FIELDVALUE(XYM_TXN_AGGREGATE_COMPLETE, "Aggregate Complete")
-        CASE_FIELDVALUE(XYM_TXN_AGGREGATE_BONDED, "Aggregate Bonded")
-        CASE_FIELDVALUE(XYM_TXN_HASH_LOCK, "Hash Lock")
-        default:
-            SNPRINTF(dst, "%s", "Unknown");
+    if (field->id == XYM_UINT16_TRANSACTION_TYPE ||
```
Comment (thuhc) [inline src/xym/format/format.c:128]: Move bottom and use "else"

**Comment by thuhc** [comment]
Change commit msg:

`Add account address/mosaic/operation restriction functions`

File: src/xym/format/format.c (hunk 1 above)
Comment (TriPhan96) [reply src/xym/format/format.c]: Fixed

File: src/xym/format/format.c [hunk 2]
```diff
@@ -211,6 +255,10 @@ static field_formatter_t get_formatter(const field_t *field) {
             return int8_formatter;
         case STI_UINT8:
             return uint8_formatter;
+        case STI_UINT8_ADDITION:
```
Comment (thuhc) [inline src/xym/format/format.c:258]: Define STI_UINT8_CUSTOM (don't need two cases ADD + DEL)

**Comment by thuhc** [comment]
@TriPhan96, please add testcases too

**Review by nanhson — APPROVED** [review]

**Review by nanhson — APPROVED** [review]

**Review by nsonanh — APPROVED** [review]

**Pull request merged**